GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -698,6 +699,61 @@ public GVRSceneObject getChildByIndex(int index) {
      */
     public Iterable<GVRSceneObject> children() {
         return new Children(this);
+    }    
+
+    /**
+     * Add a {@link GVRBaseSensor} node to the object.
+     * 
+     * Note that the attached sensor affects this node and its descendants.
+     * 
+     * @param sensor
+     *            The {@link GVRBaseSensor} to be added to the object node.
+     *            
+     * @return <code>true</code> if the sensor was added successfully;
+     *         <code>false</code> otherwise, this could possibly be due to the
+     *         {@link GVRInputManager} not being set.
+     */
+    public boolean addSensor(GVRBaseSensor sensor) {
```

Comment:
I've tried to stay in line with the User Interactivity concepts defined by X3D:

http://www.web3d.org/documents/specifications/19775-1/V3.2/Part01/components/pointingsensor.html

Since GVRf does not define an explicit grouping node, we make use of the Hybrid SceneObject node to define groups and hierarchies for User Interactivity.